src/components/ApplicationsModal/ApplicationsModal.tsx

```diff
@@ -960,7 +980,10 @@ export const ApplicationsModal: React.FC<ApplicationsModalProps> = ({
 
               <div
                 className={classes.closeModalButtonContainer}
-                onClick={() => checkNextAction("toggleModal")}
+                onClick={() => history.location.state?.fromSubsModal
```

Comment:
This ternary is being repeated some times thought the code, better to create a function to make it easier to maintain and update